### PR TITLE
ISSD-118 [CP 2.0] Create mega menu

### PR DESCRIPTION
**Pull request by liaoethan**

### CODE REVIEW

**Comment by liaoethan** [comment]
For some reason, I'm not able to add reviewers or assign people. The pull is now ready for review
@kevenleone 
@kylembischof 
@jotchon 
@Aline160 
@pedroliveirav 
@samuelleolopes 

Reviewer Notes

To test,

1. Deploy and re-run site initializer to get updated nav, custom fields, and permissions

2. Upload icons (can obtain them from figma)

3. Create a Navigation Menu

image


4. To configure the Navigation Menu (according to figma):

Example (for the first section):

**Top-level**

Resources

 

**Secondary**

Productions Documentation

Update Submenu Column Span: 4

 

**Tertiary**

Knowledge Base

Description: Articles on recurring topics, issues, and themes

Image URL: [link to icon]

Documentation

Product guides, tutorials and reference docs

Image URL: [link to icon]

Let me or @jotchon if you have any questions, thanks!

**Comment by kevenleone** [comment]
Thank you @liaoethan and @jotchon we can take a look and review :) 

Good job guys 🎉

**Review by kylembischof — COMMENTED** [review]
Just a few small formatting things and questions from me. 

Awesome job Ethan and Johanna at your teamwork at completing this. You two did it pretty quick!

Samuel asked Jose Abelenda to get you added to the reviewers list so you can request people to review in future PRs

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/group/src/customer-portal/fragments/public-sites-navigation/index.html [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+[#if configuration.darkNav]
+	[#assign navClass = "dark-nav" /]
+[/#if]
+
+<div class="public-sites-navigation-padding"></div>
+<nav class="${navClass} d-flex nav flex-column public-sites-navigation">
+	<div class="utility-nav-container">
+		<div class="align-items-center d-flex justify-content-between py-1 utility-nav w-100">
+			<div class="d-flex sites-languages-container">
+<div class="d-flex p-2 sites" data-toggle="liferay-dropdown" id="liferay-sites-id" tabindex="6">
```
[the remark this answers is not in the document]
Comment (liaoethan) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/group/src/customer-portal/fragments/public-sites-navigation/index.html:17]: I'll fix the indentations - the issue probably happened when copying it over.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/layout-page-templates/master-pages/home/page-definition.json [hunk 2]
```diff
@@ -3,671 +3,603 @@
...
-						"borderWidth": "0",
-						"shadow": "",
-						"textAlign": "",
-						"width": "100%"
-					},
-					"indexed": true,
-					"layout": {
-						"borderColor": "colorBrandPrimaryLighten6"
-					}
+				  "fragmentStyle": {
```
[the remark this answers is not in the document]
Comment (liaoethan) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/layout-page-templates/master-pages/home/page-definition.json:6]: My default tabs and spacing is different for different file types I noticed. I'll adjust that and keep indentations consistent in the future.

**Comment by liaoethan** [comment]
Thanks for the review @kylembischof , I will get in those changes in by EOD

**Comment by kylembischof** [comment]
Looks good to me. @kevenleone can you please double check since this is a big feature? :)

**Comment by liaoethan** [comment]
I spoke with @pedroliveirav earlier today since he was doing some testing. There are two changes that I need to additionally make:
1) "Home" top-level item to not have the dropdown functionality. This will be a change for the navigation_primary ADT
2) The nav currently only expands to 1920px widest, but it should accommodate for wider screens

**Comment by liaoethan** [comment]
Hi @pedroliveirav, can you please test the changes that I made? Let me know if I need to do anything additional, thanks!

**Comment by pedroliveirav** [comment]
Hi @liaoethan, I tested it again after the updates you made, but I found a strange behavior.

The template starts broken because it is not linked to some navigation menu, when I link the template with an empty navigation menu it does not break, but when I try to add items in the navigation menu, it breaks again. 

https://github.com/is-solutions-delivery/liferay-portal/assets/84352995/6daad207-378a-4d27-a9a5-0cf5ccb137b2

**Comment by liaoethan** [comment]
Thanks for catching that @pedroliveirav - this pull should be ready for review from you now @kevenleone.

**Comment by kevenleone** [comment]
LGTM

**Comment by kevenleone** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:test:relevant

**Comment by kevenleone** [comment]
ci:forward

**Comment by liaoethan** [comment]
ci:test:sf

**Comment by kylembischof** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward:force

**Comment by kylembischof** [comment]
ci:forward:force

**Comment by kevenleone** [comment]
ci:forward:force

**Comment by kylembischof** [comment]
awesome, thanks for adding to site initializer @liaoethan

**Comment by kylembischof** [comment]
@liaoethan this looks good to me, anything you were needing to add before we forward the PR?

**Comment by liaoethan** [comment]
Nope, this PR is good to go. thanks @kylembischof

**Comment by kylembischof** [comment]
ci:forward

**Comment by kylembischof** [comment]
ci:forward:force

**Comment by liaoethan** [comment]
Sorry about that, added missing closing div tag here: https://github.com/is-solutions-delivery/liferay-portal/pull/5085

**Pull request closed without merging**